src/wallet/rpcwallet.cpp

```diff
@@ -31,7 +31,14 @@
 
 CWallet *GetWalletForJSONRPCRequest(const JSONRPCRequest& request)
 {
-    // TODO: Some way to access secondary wallets
+    if (!request.wallet.empty()) {
+        for (const auto& wallet : ::vpwallets) {
+            if (request.wallet == wallet->GetName()) {
+                return wallet;
+            }
+        }
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Requested wallet does not exist or is not loaded");
```

Comment:
Reason for throwing here instead of returning `nullptr` is to make it an error to pass an invalid wallet filename to RPC calls like `validateaddress`, `createmultisig`, or `getinfo` where having a wallet is optional.